### ISSUE TITLE
Restrict users to locales they are permitted to translate in

### ISSUE THREAD

**Issue by liam-mulhall** [phetsims/rosetta#301]
Per the 2022-07-22 Rosetta 2.0 design meeting, we decided it would be convenient if translators didn't have to scroll through a list of locales that they don't translate in. Ideally, a Spanish translator would only see variants of Spanish they are comfortable translating in the locale select.

This is not going to be added in Rosetta 2.0. @mattpen needs to add locale info to user data before we can do this. See https://github.com/phetsims/website/issues/1367.

**Comment by liam-mulhall** [comment]
Similar/duplicate of https://github.com/phetsims/rosetta/issues/71. #71 has two suggestions, so I think I would prefer to keep this issue.

**Comment by liam-mulhall** [comment]
Related to https://github.com/phetsims/rosetta/issues/109.

**Comment by liam-mulhall** [comment]
I'm unassigning myself since I'm not working on this anytime soon.

**Comment by liammulh** [comment]
Notes from Jan 20 meeting:

- MP: We need a strong concept of what we want for the translator's locale.
- MP: We could provide the translator's locale in the check-login object.

**Comment by mattpen** [comment]
@liammulh - The translator's locales are now provided in the check-login response as the `translatorLocales` property.  To start with, if this value is null or an empty string we'll need to provide access to all locales, as we can't backfill all of the trusted translators with approved locales immediately.  But eventually, this will contain a comma separated list of locale codes.

You can test this out by adding locales to your website user or any test users that you have by signing into the website, going to Administration > Users > and using the "Edit User Privileges" form.
